aptos-move/move-examples/marketplace/sources/fee_schedule.move

```diff
@@ -0,0 +1,498 @@
+        /// Fixed rate for bidding
+        bidding_fee: u64,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Fixed rate for listing
+    struct FixedRateListingFee has drop, key {
+        /// Fixed rate for listing
+        listing_fee: u64,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Fixed rate for commission
+    struct FixedRateCommission has drop, key {
+        /// Fixed rate for commission
+        commission: u64,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Percentage-based rate for commission
+    struct PercentageRateCommission has drop, key {
+        /// Denominator for the commission rate
+        denominator: u64,
+        /// Numerator for the commission rate
+        numerator: u64,
+    }
+
+    /// Event representing a change to the marketplace configuration
+    struct MutationEvent has drop, store {
+        /// The type info of the struct that was updated.
+        updated_resource: String,
+    }
+
+    // Initializers
+
+    /// Create a marketplace with a fixed bidding and listing rate and a percentage commission.
+    public entry fun init(
+        creator: &signer, 
+        fee_address: address,
+        bidding_fee: u64,
+        listing_fee: u64,
+        commission_denominator: u64,
+        commission_numerator: u64,
+    ) {
+        init_internal(
+            creator,
+            fee_address,
+            bidding_fee,
+            listing_fee,
+            commission_denominator,
+            commission_numerator,
+        );
+    }
+
+
+    public fun init_internal(
+        creator: &signer, 
+        fee_address: address,
+        bidding_fee: u64,
+        listing_fee: u64,
+        commission_denominator: u64,
+        commission_numerator: u64,
+    ): Object<FeeSchedule> {
+        assert!(
+            commission_numerator <= commission_denominator,
+            error::invalid_argument(EEXCEEDS_MAXIMUM),
+        );
+        assert!(
+            commission_denominator != 0,
+            error::out_of_range(EDENOMINATOR_IS_ZERO),
+        );
+
+        let (constructor_ref, fee_schedule_signer) = empty_init(creator, fee_address);
+        move_to(&fee_schedule_signer, FixedRateBiddingFee { bidding_fee });
+        move_to(&fee_schedule_signer, FixedRateListingFee { listing_fee });
+        let commission_rate = PercentageRateCommission {
+            denominator: commission_denominator,
+            numerator: commission_numerator,
+        };
+        move_to(&fee_schedule_signer, commission_rate);
+        object::object_from_constructor_ref(&constructor_ref)
+    }
+
+    /// Create a marketplace with no fees.
+    public entry fun empty(
+        creator: &signer,
+        fee_address: address,
+    ) {
+        empty_init(creator, fee_address);
+    }
+
+    inline fun empty_init(creator: &signer, fee_address: address): (ConstructorRef, signer) {
```

Comment:
Related to my earlier comment in listing.move, it might be better to not have FeeSchedule then. That might be easier to index and query than having FeeSchedule with none of the individual fee resources (which are also confusing and tricky to deal with indexing and off chain wise)